File: Problems/No.219_Contains_Duplicate_II/contains_duplicate_II.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """

        # parameter
        Nmapping = {}

        # start computing
        for i in range(len(nums)):
            if ((Nmapping.get(nums[i]) != None) and (i-Nmapping.get(nums[i]) <= k)):
                return True
            Nmapping[nums[i]] = i

        return False

File: Problems/No.219_Contains_Duplicate_II/contains_duplicate_II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        
        // parameter
        std::unordered_map<int, int> Nmapping;

        // start computing
        for (int i=0; i<nums.size(); i++) {

            if ((Nmapping.find(nums[i]) != Nmapping.end()) && 
                (i-Nmapping[nums[i]] <= k)) return true;
            Nmapping[nums[i]] = i;

        }

        return false;

    }
};
